src/eth1/impl/ethers.ts

```diff
@@ -1,52 +1,62 @@
 import BN from "bn.js";
 import {EventEmitter} from "events";
-import {ethers} from "ethers";
+import {Contract, ethers} from "ethers";
 import {deserialize} from "@chainsafe/ssz";
 
-import {bytes32, Deposit, DepositData, Eth1Data} from "../types";
+import {bytes32, Deposit, DepositData, Eth1Data, number64} from "../../types";
 
-import {Eth1Notifier, Eth1Options} from "./interface";
-import logger from "../logger";
-import {isValidAddress} from "../util/address";
+import {Eth1Notifier, Eth1Options} from "../interface";
+import logger from "../../logger";
+import {isValidAddress} from "../../util/address";
+import {DB} from "../../db";
 
 export interface EthersEth1Options extends Eth1Options {
   provider: ethers.providers.BaseProvider;
+  db: DB;
+  contract?: Contract;
```

Comment:
I don't think this contract ever gets a chance to be used in the current logic. `initContract` always creates a `Contract` from scratch.

Reply:
This allows me to override contract with stub in tests.
https://github.com/ChainSafe/lodestar/pull/180/files/40d22e3e3004d8c2238eaa2317a024da5f87fef9#diff-7f07d5ce2a5caff843f94228e726bde3R51